File: v1/src/main/java/com/api/v1/services/payment/AllPaymentsRetrievalService.java
package com.api.v1.services.payment;

import com.api.v1.dtos.payment.PaymentResponseDto;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface AllPaymentsRetrievalService {

    Flux<PaymentResponseDto> findAll();
    Mono<PaymentResponseDto> findByOrderNumber(String orderNumber);

}

File: kotlin/v2/src/main/kotlin/com/api/v2/payment/services/AllPaymentsRetrievalService.kt
package com.api.v2.payment.services

import com.api.v2.payment.dtos.PaymentResponseDto
import kotlinx.coroutines.flow.Flow

interface AllPaymentsRetrievalService {

    suspend fun findAll(): Flow<PaymentResponseDto>
    suspend fun findByOrderNumber(orderNumber: String): PaymentResponseDto

}